### DISCUSSION TITLE
Let's talk about Mojolicious's favorite features

### DISCUSSION THREAD

**Discussion by yuki-kimoto** [mojolicious/mojo#1652] [General]
Hi, do you have a favorite feature of Mojolicious?

I like url_with method.

```
<%= url_with->query({page => 2}) %>
```

I can easily write pagination!

**Comment by mche** [comment]
The Mojolicious is whole my favorite feature!

**Comment by kberov** [comment]
The possibility to start small even on a cheap shared hosting as a CGI application and then just move to a standalone application.
The possibility to switch from SQLite to PostgreSQL/MySQL when I need it.
So the most regarded by me feature is **flexibility**.

**Comment by kraih** [comment]
My favorite features are probably `Mojo::DOM` and `Minion`.

**Comment by s1037989** [comment]
Mojo::Base -- I use it in *everything*!
Minion
Mojo::Pg (and more lately Mojo::SQLite)
Mojo::*, such as Mojo::IOLoop, Mojo::Promise, Mojo::JSON, Mojo::Collection, Mojo::File, Mojo::Util::*, Mojo::Loader
The complete, detailed, and clear documentation, and the core team's commitment to keeping it solid.
The core team's discipline in ensuring good quality code, and development principles.
The plethora of tests to ensure a rock-solid product.

This toolkit and framework are **amazing**!  But more than that, I have learned so much about coding, advanced software engineering, design, discipline, techniques, etc.  I follow this project like a hawk and am able to apply so much from it to everything else in my software development career, even things in other languages (e.g. my big project right now is in Bash and I built myself an Event Loop inspired by Mojo::IOLoop).  My only regret is that I am unable to keep up with the community via IRC.  I unfortunately am unable to connect to IRC at times in the day convenient to anyone else, so it's always been quiet when I've connected.

**Comment by yuki-kimoto** [comment]
Thank you for your comments.

I also learned a lot from the Mojolicious code.

Perl had been said to be a writing-only language.

Mojolicious made me realize that it was a misunderstanding.

Mojolicious is flexible, and there are all the necessary and sufficient functions for web development.

I also enjoy using Mojo :: DOM.

```
my $title = $dom->at('.title')->next->all_text;
```
